drivers/src/preconditioned_key.rs

```diff
@@ -0,0 +1,70 @@
+/*++
+
+Licensed under the Apache-2.0 license.
+
+File Name:
+
+    preconditioned_key.rs
+
+Abstract:
+
+    A FIPS-approved method for combining keys together leveraging an HMAC-based key-extraction process.
+
+--*/
+
+use crate::{
+    hmac_kdf, Aes, AesKey, AesOperation, Hmac, HmacData, HmacKey, HmacMode, HmacTag, KeyReadArgs,
+    KeyUsage, KeyWriteArgs, Trng,
+};
+
+use caliptra_error::{CaliptraError, CaliptraResult};
+
+pub fn preconditioned_key_extract(
+    input_key: HmacKey,
+    output_key: HmacTag,
+    kdf_label: &[u8],
+    salt: HmacKey,
+    trng: &mut Trng,
+    hmac: &mut Hmac,
+    aes: &mut Aes,
+) -> CaliptraResult<()> {
+    let HmacTag::Key(output_kv) = output_key else {
+        Err(CaliptraError::RUNTIME_DRIVER_PRECONDITIONED_KEY_INVALID_INPUT)?
+    };
+
+    let mut aes_key = [0; 32];
+    let aes_key = match salt {
+        HmacKey::Array4x16(arr) => {
+            let arr: [u8; 64] = arr.into();
+            aes_key.clone_from_slice(&arr[0..32]);
+            AesKey::Array(&aes_key)
```

Comment:
I suppose it doesn't matter as long as we're consistent, but I think the the HMAC keys come out with the words byte-reversed from what AES expects.

Reply:
Good to know - I think this will almost always be a KV vault, but I will add a comment here in a separate PR